src/libraries/System.Runtime.Loader/tests/GenerateHotReloadDelta.targets

```diff
@@ -0,0 +1,81 @@
+<Project>
+  <!-- Invoke hotreload-delta-gen to apply the DeltaScript to the output assembly to
+       create the .dmeta, .dil and .dpdb files on the OutputPath -->
+  <!-- Have to be careful about AfterTargets. If we happen to go after a design-time target, we could get into an infinite loop. -->
+  <Target Name="CompileDiff" AfterTargets="Build" Condition="'$(DeltaScript)' != ''">
+    <Message Importance="High" Condition="'$(HotreloadDeltaGenFullPath)' == ''" Text="WARNING: HotreloadDeltaGenFullPath property is unset.  Will run 'hotreload-delta-gen' assuming it is on the PATH" />
+    <PropertyGroup>
+      <HotreloadDeltaGenFullPath Condition="'$(HotreloadDeltaGenFullPath)' == ''">hotreload-delta-gen</HotreloadDeltaGenFullPath>
+      <HotreloadDeltaGenArgs>-msbuild:$(MSBuildProjectFullPath)</HotreloadDeltaGenArgs>
+      <!-- HACK: have to pass config and RID  so that this target works for a 'dotnet publish' run.
+	         What other properties do  I need to pass? Maybe hotreload-delta-gen should just expose an MSBuild task so we can pass everything -->
+      <HotreloadDeltaGenArgs Condition="'$(Configuration)' != ''">$(HotreloadDeltaGenArgs) -p:Configuration=$(Configuration)</HotreloadDeltaGenArgs>
+      <HotreloadDeltaGenArgs Condition="'$(RuntimeIdentifier)' != ''">$(HotreloadDeltaGenArgs) -p:RuntimeIdentifier=$(RuntimeIdentifier)</HotreloadDeltaGenArgs>
+      <HotreloadDeltaGenArgs Condition="'$(BuiltRuntimeConfiguration)' != ''">$(HotreloadDeltaGenArgs) -p:BuiltRuntimeConfiguration=$(BuiltRuntimeConfiguration)</HotreloadDeltaGenArgs>
+      <HotreloadDeltaGenArgs>$(HotreloadDeltaGenArgs) -script:$(DeltaScript)</HotreloadDeltaGenArgs>
+    </PropertyGroup>
+    <Exec Command="$(HotreloadDeltaGenFullPath) $(HotreloadDeltaGenArgs)"/>
+  </Target>
+
+  <!-- Computes the names of the files that hotreload-delta-gen will produce, given the name of the base assembly and the
+       number of deltas -->
+  <UsingTask TaskName="ComputeDeltaFileOutputNames" TaskFactory="RoslynCodeTaskFactory"
```

Comment:
Can we actually create a tasks assembly to use here (probably in `src/tasks`) instead of using the RoslynCodeTaskFactory?

Reply:
Added a `GenerateHotReloadDelta.BuildTasks` assembly next to the tests.  Also took the opportunity to compute the number of expected deltas directly from the json script, instead of requiring a (redundant) DeltaCount property in the projects. Should  improve usability a bit.